### PR TITLE
Migrate to Relay Modern

### PR DESCRIPTION
**Pull request by alloy**
Here be 🐉

### CODE REVIEW

**Comment by alloy** [comment]
Ugh… that list of conflicts :sob:

File: src/Components/Onboarding/Steps/Artists/ItemLink.tsx [hunk 1]
```diff
@@ -38,28 +39,71 @@ const Col = styled.div`
...
+}
 
 class ItemLink extends React.Component<Props, State> {
   state = {
     fadeIn: false,
     fadeOut: false,
+    artist_id: this.props.artist.id,
+  }
+
+  followArtist() {
```
Comment (alloy) [inline src/Components/Onboarding/Steps/Artists/ItemLink.tsx]: @xtina-starr @mzikherman Here’s the initial setup for the follow artist mutation.

File: src/Components/Onboarding/Steps/Artists/ItemLink.tsx [hunk 2]
```diff
@@ -38,28 +39,71 @@ const Col = styled.div`
...
+    console.log("follow artist!")
+    const storeUpdater: SelectorStoreUpdater = (store, suggestedArtistData) => {
+      // Replace now followed artist in list with newly suggested artist
```
Comment (alloy) [inline src/Components/Onboarding/Steps/Artists/ItemLink.tsx]: This is where you’d replace the followed artist by manipulating the store.

File: src/Components/Onboarding/Steps/Artists/ItemLink.tsx [hunk 3]
```diff
@@ -38,28 +39,71 @@ const Col = styled.div`
...
+    }
+
+    commitMutation(this.props.relay.environment, {
+      mutation: graphql`
+        mutation ItemLinkFollowArtistMutation($input: FollowArtistInput!) {
+          followArtist(input: $input) {
+            artist {
+              # Get related artist (not this field that’s currently used!)
+              artists(size: 1) {
```
Comment (alloy) [inline src/Components/Onboarding/Steps/Artists/ItemLink.tsx]: Use the real (still to be implemented?) suggested artist field here.

File: src/Components/Onboarding/Steps/Artists/ItemLink.tsx [hunk 4]
```diff
@@ -38,28 +39,71 @@ const Col = styled.div`
...
+                name
+              }
+            }
+          }
+        }
+      `,
+      variables: {
+        input: {
+          artist_id: this.props.artist.id,
+          unfollow: false,
```
Comment (alloy) [inline src/Components/Onboarding/Steps/Artists/ItemLink.tsx]: Such negation… this param should really be called `follow` imo, but that’s probably a breaking schema change.

**Comment by orta** [comment]
good luck with that merge :P

File: typings/react.d.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from "react"
+
+declare module "react" {
+  /**
+   * This typing only exists in the React 16 typings, but the storybook-react typings assume we’re already on that even
+   * though React 16 is not a hard dependency of storybook-react. Rather than updating to React 16, we simply augment
+   * the React 15 typings to add the missing type.
+   */
+  type ComponentType<P = {}> = React.ComponentClass<P> | React.StatelessComponent<P>;
+}
```
Comment (alloy) [inline typings/react.d.ts]: Interestingly enough there actually _is_ a React 15 version of the typings that has this type, but when I try to install that all the other typings packages related to react install their own copy of the react typings, which breaks everything https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b13b82c214ac5ed2397ec5f3ab6a157a9fc7e220/types/react/v15/index.d.ts#L78.

**Comment by mzikherman** [comment]
Alright, all the updates are here. Force PR is ready to go once we publish a new version.

@kanaabe is standing behind me and approves.

**Pull request merged**